contracts/exchange/test/utils/match_order_tester.ts

```diff
@@ -108,37 +129,117 @@ export class MatchOrderTester {
         exchangeWrapper: ExchangeWrapper,
         erc20Wrapper: ERC20Wrapper,
         erc721Wrapper: ERC721Wrapper,
         erc1155ProxyWrapper: ERC1155ProxyWrapper,
+        batchMatchOrdersCallAsync?: BatchMatchOrdersAsyncCall,
+        batchMatchOrdersWithMaximalFillCallAsync?: BatchMatchOrdersAsyncCall,
         matchOrdersCallAsync?: MatchOrdersAsyncCall,
+        matchOrdersWithMaximalFillCallAsync?: MatchOrdersAsyncCall,
     ) {
         this.exchangeWrapper = exchangeWrapper;
         this.erc20Wrapper = erc20Wrapper;
         this.erc721Wrapper = erc721Wrapper;
         this.erc1155ProxyWrapper = erc1155ProxyWrapper;
+        this.batchMatchOrdersCallAsync = batchMatchOrdersCallAsync;
+        this.batchMatchOrdersWithMaximalFillCallAsync = batchMatchOrdersWithMaximalFillCallAsync;
         this.matchOrdersCallAsync = matchOrdersCallAsync;
+        this.matchOrdersWithMaximalFillCallAsync = matchOrdersWithMaximalFillCallAsync;
         this._initialTokenBalancesPromise = this.getBalancesAsync();
     }
 
+    /**
+     * Performs batch order matching on a set of complementary orders and asserts results.
+     * @param orders The list of orders and filled states
+     * @param matchPairs An array of left and right indices that will be used to perform
+     *                   the expected simulation.
+     * @param takerAddress Address of taker (the address who matched the two orders)
+     * @param expectedTransferAmounts Expected amounts transferred as a result of each round of
+     *                                order matching. Omitted fields are either set to 0 or their
+     *                                complementary field.
+     * @param withMaximalFill A boolean that indicates whether the "maximal fill" order matching
+     *                        strategy should be used.
+     * @return Results of `batchMatchOrders()`.
+     */
+    public async batchMatchOrdersAndAssertEffectsAsync(
+        orders: BatchMatchedOrders,
+        takerAddress: string,
+        matchPairs: Array<[number, number]>,
+        expectedTransferAmounts: Array<Partial<MatchTransferAmounts>>,
+        withMaximalFill: boolean,
+        initialTokenBalances?: TokenBalances,
+    ): Promise<BatchMatchResults> {
+        // Ensure that the provided input is valid.
+        expect(matchPairs.length).to.be.eq(expectedTransferAmounts.length);
+        expect(orders.leftOrders.length).to.be.eq(orders.leftOrdersTakerAssetFilledAmounts.length);
+        expect(orders.rightOrders.length).to.be.eq(orders.rightOrdersTakerAssetFilledAmounts.length);
+        // Ensure that the exchange is in the expected state.
+        await assertBatchOrderStatesAsync(orders, this.exchangeWrapper);
+        // Get the token balances before executing `batchMatchOrders()`.
+        const _initialTokenBalances = initialTokenBalances
+            ? initialTokenBalances
+            : await this._initialTokenBalancesPromise;
+        // Execute `batchMatchOrders()`
```

Comment:
Around here we need to call `getBatchMatchOrders...ResultsAsync()` and assert the results against our simulation.

Ditto for the `matchOrdersAndAssertEffectsAsync()` function.